src/connection.rs

```diff
@@ -512,6 +492,13 @@ impl PubSub {
     }
 }
 
+impl<'a> Drop for PubSub<'a> {
+    fn drop(&mut self) {
+        let _ = cmd("UNSUBSCRIBE").query::<()>(self.con);
+        let _ = cmd("PUNSUBSCRIBE").query::<()>(self.con);
```

Comment:
Unsubscribing will trigger a message per channel that's unsubscribed.
Will this cause problems for the underlying connection if it is later reused?
Having a test for this would be great.

Reply:
It shouldn't because `query` will already expect to receive a result from the server and parse it. I agree that we should have a test for this including that the connection is no longer subscribed to anything.

Reply:
As it turns out, a message per channel actually means a distinct message per channel at a protocol level. This will take a bit of work to support. Nice catch!